ad_block_client.cc

```diff
@@ -199,13 +200,16 @@ void parseFilter(const char *input, const char *end, Filter *f,
     BloomFilter *exceptionBloomFilter,
     HashSet<Filter> *hostAnchoredHashSet,
     HashSet<Filter> *hostAnchoredExceptionHashSet,
-    HashSet<CosmeticFilter> *simpleCosmeticFilters) {
+    HashSet<CosmeticFilter> *simpleCosmeticFilters,
+    bool preserveRules) {
   FilterParseState parseState = FPStart;
   const char *p = input;
+  const char *q = p;
   char data[kMaxLineLength];
   memset(data, 0, sizeof data);
   int i = 0;
-
+  int ruleTextLength = 0;
```

Comment:
I think we can just subtract `filterRuleEndPos` from the line start pos to get the same length. So we can probably get rid of this.